hack/update-codegen.sh

```diff
@@ -15,6 +15,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+# TODO: migrate to k8s.io/code-generator/kube_codegen.sh
```

Comment:
both generate-groups and generate-internal-groups seems to be gone with v1.30.

Reply:
Could you please resolve this TODO? What does "gone" mean for this script?

Typically, I try to follow this: https://github.com/kubernetes/sample-controller/blob/master/hack/update-codegen.sh

Reply:
`generate-internal-groups.sh` and `generate-groups.sh` which this script depends have been deprecated since 1.29 and removed in 1.30.

https://github.com/kubernetes/code-generator/blob/3f91291add4375b42d0f8c4d6d041bb6ed440d97/generate-groups.sh#L52-L54

The recommendation is to switch to `kube_codegen.sh` instead.

Happy to include the change here, just wanted to keep the scope small.

Reply:
so I ended up updating the k8s deps in the process including controller-runtime and the change is here (+boilerplate changes):

https://github.com/admiraltyio/admiralty/commit/bf71f85547252a2b64a842e3d63954eb45f0e312

Thoughts on having this as a separate follow-up PR? I guess it doesn't matter much since I noticed PRs aren't merged squashed but it would be easier to review the diff separately. Happy to do either, let me know.